### PR TITLE
my hw jv-dependency-injection

### PR DESCRIPTION
**Pull request by VolodymyrKachan**

### CODE REVIEW

File: src/main/java/mate.academy/lib/Injector.java [hunk 1]
```diff
@@ -1,13 +1,75 @@
...
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (NoSuchMethodException | InvocationTargetException | InstantiationException
+                 | IllegalAccessException e) {
```
Comment (valentyn-chepiha) [inline src/main/java/mate.academy/lib/Injector.java:60]:
```suggestion
        } catch (ReflectiveOperationException e) {
```

**Review by valentyn-chepiha — CHANGES_REQUESTED** [review]
Good job

**Review by valentyn-chepiha — APPROVED** [review]

File: src/main/java/mate.academy/lib/Injector.java [hunk 2]
```diff
@@ -1,13 +1,73 @@
...
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field filed : declaredFields) {
+            if (filed.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = injector.getInstance(filed.getType());
+                clazzImplementationInstance = createNewInstance(clazz);
+                try {
+                    filed.setAccessible(true);
+                    filed.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't initialize filed value. "
```
Comment (Noct2000) [inline src/main/java/mate.academy/lib/Injector.java:38]: add root cause exception

**Review by Noct2000 — CHANGES_REQUESTED** [review]

File: src/main/java/mate.academy/lib/Injector.java [hunk 3]
```diff
@@ -1,13 +1,73 @@
...
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Injection failed. @Component annotation is missed in "
+                    + interfaceClazz.getName());
```
Comment (Noct2000) [inline src/main/java/mate.academy/lib/Injector.java:27]:
```suggestion
                    + clazz.getName());
```

**Review by Noct2000 — CHANGES_REQUESTED** [review]

**Review by Noct2000 — APPROVED** [review]

**Pull request closed without merging**